src/renderers/dom/server/ReactServerUpdateQueue.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactServerUpdateQueue
+ * @flow
+ */
+
+'use strict';
+
+var ReactUpdateQueue = require('ReactUpdateQueue');
+var Transaction = require('Transaction');
+var warning = require('warning');
+
+import type React from 'react';
+
+function warnNoop(publicInstance: React.Component<*,*,*>, callerName: string) {
```

Comment:
Let’s type it as `ReactComponent<any, any, any>`. That’s what [we already use](https://github.com/facebook/react/blob/6cc037bd0d49f1f55cbdd36fbfd395118c64d393/src/renderers/native/ReactNative.js#L31) in this repo.